## E-commerce Sales Analysis

#### Data Analysis using Python & Pandas

This notebook analyzes an e-commerce sales dataset using Python and Pandas.
The analysis focuses on sales, quantity, categories, cities, products,
payment methods, and other important sales statistics.

## 🎯 Objectives

The main objectives of this analysis are:

- Load and understand the e-commerce sales dataset
- Perform data selection and filtering
- Sort data to identify important orders
- Analyze sales using `groupby()` and aggregation
- Compare categories, cities, products, and payment methods
- Identify top-performing products and categories
- Summarize meaningful insights from the analysis

## 1. Loading the Dataset

In [2]:
import pandas as pd
print("Pandas imported successfully!")

Pandas imported successfully!


In [3]:
df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

df.head()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


## 2. Understanding the Dataset

#### Check the shape of the dataset

In [4]:
df.shape

(100, 13)

#### Check the shape of the dataset

In [5]:
df.columns

Index(['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category',
       'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales',
       'Rating', 'Payment_Method'],
      dtype='str')

#### Check data types and missing values

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    str    
 2   Customer          100 non-null    str    
 3   City              100 non-null    str    
 4   Region            100 non-null    str    
 5   Category          100 non-null    str    
 6   Product           100 non-null    str    
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12  Payment_Method    100 non-null    str    
dtypes: float64(2), int64(4), str(7)
memory usage: 10.3 KB


#### Basic statistical summary

In [7]:
df.describe()

,Order_ID,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating
count,100.000000,100.000000,100.000000,100.000000,100.000000,100.000000
mean,1050.500000,2.940000,1647.000000,6.500000,4472.629500,4.322000
std,29.011492,1.391315,970.054669,6.092718,3324.306691,0.414285
min,1001.000000,1.000000,499.000000,0.000000,449.100000,3.500000
25%,1025.750000,2.000000,799.000000,0.000000,1948.050000,4.100000
50%,1050.500000,3.000000,1499.000000,5.000000,3228.300000,4.300000
75%,1075.250000,4.000000,2499.000000,11.250000,6427.562500,4.700000
max,1100.000000,5.000000,3499.000000,15.000000,13995.000000,5.000000


#### Check first and last few records

In [8]:
df.head()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [9]:
df.tail()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card


## 3. Data Selection

#### Selecting a Single Column

In [10]:
df["Product"]

0         Wireless Headphones
1               Running Shoes
2                  Power Bank
3         Mechanical Keyboard
4                 Smart Watch
               ...           
95               Dumbbell Set
96                    T-Shirt
97            Electric Kettle
98    Machine Learning Basics
99      Data Science Handbook
Name: Product, Length: 100, dtype: str

In [11]:
df["Total_Sales"]

0     4497.00
1     2799.00
2     2038.30
3     9496.20
4     3134.05
       ...   
95    9495.25
96     679.15
97    5996.00
98    1998.00
99    4495.00
Name: Total_Sales, Length: 100, dtype: float64

#### Select multiple columns

In [12]:
df[["Order_ID", "Product", "Quantity", "Total_Sales"]]

,Order_ID,Product,Quantity,Total_Sales
0,1001,Wireless Headphones,3,4497.00
1,1002,Running Shoes,1,2799.00
2,1003,Power Bank,2,2038.30
3,1004,Mechanical Keyboard,4,9496.20
4,1005,Smart Watch,1,3134.05
...,...,...,...,...
95,1096,Dumbbell Set,5,9495.25
96,1097,T-Shirt,1,679.15
97,1098,Electric Kettle,4,5996.00
98,1099,Machine Learning Basics,2,1998.00


#### Select customer and city details

In [15]:
df[["Customer", "City", "Region"]]

,Customer,City,Region
0,Kabir Ali,Hyderabad,South
1,Aditya Verma,Jammu,North
2,Sara Ahmed,Chennai,South
3,Kabir Ali,Lucknow,North
4,Priya Menon,Jammu,North
...,...,...,...
95,Aditya Verma,Kochi,South
96,Arjun Nair,Mumbai,West
97,Yusuf Dar,Bengaluru,South
98,Hiba Sheikh,Ahmedabad,West


#### Select rows using loc[]

In [18]:
# To select First row
df.loc[0]

Order_ID                           1001
Order_Date                   2026-02-27
Customer                      Kabir Ali
City                          Hyderabad
Region                            South
Category                    Electronics
Product             Wireless Headphones
Quantity                              3
Unit_Price                         1499
Discount_Percent                      0
Total_Sales                      4497.0
Rating                              3.8
Payment_Method         Cash on Delivery
Name: 0, dtype: object

In [19]:
# To select first 5 rows
df.loc[0:4]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [20]:
# Select specific rows and columns using loc[]
df.loc[0:4, ["Order_ID", "Product", "Total_Sales"]]

,Order_ID,Product,Total_Sales
0,1001,Wireless Headphones,4497.00
1,1002,Running Shoes,2799.00
2,1003,Power Bank,2038.30
3,1004,Mechanical Keyboard,9496.20
4,1005,Smart Watch,3134.05


#### Select specific rows using iloc

In [13]:
# First 5 rows
df.iloc[0:5]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [14]:
# Specific rows and columns:
df.iloc[0:5, 0:5]

,Order_ID,Order_Date,Customer,City,Region
0,1001,2026-02-27,Kabir Ali,Hyderabad,South
1,1002,2026-01-24,Aditya Verma,Jammu,North
2,1003,2026-04-18,Sara Ahmed,Chennai,South
3,1004,2026-03-13,Kabir Ali,Lucknow,North
4,1005,2026-03-30,Priya Menon,Jammu,North


## 4. Filtering Data

In [21]:
# Find orders with sales greater than ₹500
df[df["Total_Sales"] > 500]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI


In [98]:
# Find orders with quantity greater than 4
df[df["Quantity"] > 4]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
37,1038,2026-03-08,Aditya Verma,Pune,West,Sports,Dumbbell Set,5,1999,5,9495.25,5.0,Debit Card
41,1042,2026-06-20,Manya Rao,Kochi,South,Home & Kitchen,Water Bottle,5,699,0,3495.00,4.2,UPI
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
66,1067,2026-05-21,Karan Joshi,Mumbai,West,Books,The Alchemist,5,499,10,2245.50,4.0,Net Banking
71,1072,2026-05-03,Aarav Sharma,Jaipur,North,Sports,Yoga Mat,5,899,10,4045.50,4.3,Debit Card


In [30]:
# Filter a particular category
df[df["Category"] == "Electronics"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI
25,1026,2026-05-12,Sara Ahmed,Chandigarh,North,Electronics,Wireless Headphones,3,1499,0,4497.00,4.2,Net Banking
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI


In [31]:
# Filter a particular payment method
df[df["Payment_Method"] == "UPI"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
10,1011,2026-03-22,Rahul Das,Hyderabad,South,Home & Kitchen,Coffee Maker,2,3499,0,6998.00,4.2,UPI
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.00,4.6,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI
15,1016,2026-03-06,Manya Rao,Pune,West,Books,The Alchemist,4,499,15,1696.60,3.5,UPI
18,1019,2026-05-19,Reyansh Jain,Chennai,South,Clothing,Jacket,3,2499,0,7497.00,4.7,UPI
19,1020,2026-04-03,Mehak Bhat,Chennai,South,Home & Kitchen,Mixer Grinder,1,2999,0,2999.00,4.1,UPI
23,1024,2026-02-27,Rohan Mehta,Jaipur,North,Sports,Running Shoes,1,2799,0,2799.00,4.3,UPI
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI


In [32]:
# Multiple conditions
df[(df["Category"] == "Electronics") & (df["Total_Sales"] > 500)]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI
25,1026,2026-05-12,Sara Ahmed,Chandigarh,North,Electronics,Wireless Headphones,3,1499,0,4497.00,4.2,Net Banking
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI


In [33]:
# OR condition
df[(df["Category"] == "Electronics") | (df["Category"] == "Furniture")]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
14,1015,2026-06-24,Sana Malik,Kolkata,East,Electronics,Wireless Headphones,2,1499,0,2998.00,4.8,UPI
25,1026,2026-05-12,Sara Ahmed,Chandigarh,North,Electronics,Wireless Headphones,3,1499,0,4497.00,4.2,Net Banking
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI


In [34]:
# Filter high-rated orders
df[df["Rating"] >= 4]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
...,...,...,...,...,...,...,...,...,...,...,...,...,...
92,1093,2026-05-16,Aisha Mir,Hyderabad,South,Sports,Cricket Bat,4,2499,15,8496.60,4.5,Cash on Delivery
94,1095,2026-02-25,Maryam Khan,Mumbai,West,Electronics,Power Bank,2,1199,0,2398.00,4.6,Debit Card
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking


In [35]:
# Filter a particular city
df[df["City"] == "Kochi"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card
34,1035,2026-05-03,Dev Patel,Kochi,South,Home & Kitchen,Coffee Maker,1,3499,15,2974.15,4.7,Credit Card
41,1042,2026-06-20,Manya Rao,Kochi,South,Home & Kitchen,Water Bottle,5,699,0,3495.00,4.2,UPI
63,1064,2026-02-07,Aman Kumar,Kochi,South,Books,Atomic Habits,4,599,0,2396.00,3.5,UPI
74,1075,2026-05-17,Aarav Sharma,Kochi,South,Clothing,Jeans,2,1899,15,3228.30,4.5,Cash on Delivery
85,1086,2026-06-16,Arjun Nair,Kochi,South,Clothing,Jeans,2,1899,15,3228.30,4.8,Cash on Delivery
90,1091,2026-06-26,Hiba Sheikh,Kochi,South,Books,Atomic Habits,5,599,0,2995.00,5.0,Net Banking
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card


## 5. Sorting Data

In [36]:
# Sort orders by Total Sales — highest first
df.sort_values(by="Total_Sales", ascending=False)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
...,...,...,...,...,...,...,...,...,...,...,...,...,...
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
73,1074,2026-03-28,Alina Shah,Lucknow,North,Books,Atomic Habits,1,599,10,539.10,5.0,Net Banking
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery


In [37]:
# Sort by Total Sales — lowest first
df.sort_values(by="Total_Sales", ascending=True)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
62,1063,2026-04-17,Aman Kumar,Chandigarh,North,Books,The Alchemist,1,499,10,449.10,4.8,Credit Card
73,1074,2026-03-28,Alina Shah,Lucknow,North,Books,Atomic Habits,1,599,10,539.10,5.0,Net Banking
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
...,...,...,...,...,...,...,...,...,...,...,...,...,...
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card


In [38]:
# Top 5 highest sales orders
df.sort_values(by="Total_Sales", ascending=False).head(5)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking


In [39]:
# Top 5 products by quantity sold
df.sort_values(by="Quantity", ascending=False).head(5)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
37,1038,2026-03-08,Aditya Verma,Pune,West,Sports,Dumbbell Set,5,1999,5,9495.25,5.0,Debit Card
71,1072,2026-05-03,Aarav Sharma,Jaipur,North,Sports,Yoga Mat,5,899,10,4045.50,4.3,Debit Card


In [40]:
# Sort by Rating
df.sort_values(by="Rating", ascending=False).head(5)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
35,1036,2026-06-02,Rohan Mehta,Srinagar,North,Sports,Yoga Mat,2,899,0,1798.00,5.0,Credit Card
47,1048,2026-06-24,Sara Ahmed,Lucknow,North,Home & Kitchen,Coffee Maker,3,3499,0,10497.00,5.0,UPI
37,1038,2026-03-08,Aditya Verma,Pune,West,Sports,Dumbbell Set,5,1999,5,9495.25,5.0,Debit Card
55,1056,2026-04-21,Rohan Mehta,Srinagar,North,Sports,Dumbbell Set,2,1999,5,3798.10,5.0,Debit Card


In [41]:
# Sort by multiple columns
df.sort_values(by=["Category", "Total_Sales"], ascending=[True, False])

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card
29,1030,2026-03-13,Karan Joshi,Bengaluru,South,Books,Python Programming,4,749,0,2996.00,4.1,UPI
90,1091,2026-06-26,Hiba Sheikh,Kochi,South,Books,Atomic Habits,5,599,0,2995.00,5.0,Net Banking
87,1088,2026-06-06,Alina Shah,Lucknow,North,Books,Python Programming,4,749,15,2546.60,4.5,Debit Card
72,1073,2026-06-04,Sara Ahmed,Kolkata,East,Books,Atomic Habits,5,599,15,2545.75,4.6,Credit Card
...,...,...,...,...,...,...,...,...,...,...,...,...,...
39,1040,2026-05-18,Aditya Verma,Ahmedabad,West,Sports,Dumbbell Set,1,1999,0,1999.00,4.7,Debit Card
35,1036,2026-06-02,Rohan Mehta,Srinagar,North,Sports,Yoga Mat,2,899,0,1798.00,5.0,Credit Card
20,1021,2026-01-18,Reyansh Jain,Chandigarh,North,Sports,Yoga Mat,1,899,10,809.10,4.7,Credit Card
30,1031,2026-01-04,Rohan Mehta,Ahmedabad,West,Sports,Yoga Mat,1,899,15,764.15,4.1,Credit Card


## 6. Sales Statistics

In [44]:
# Total Sales
total_sales = df["Total_Sales"].sum()
total_sales
print(f"Total Sales: ₹{total_sales:.2f}")

Total Sales: ₹447262.95


In [45]:
# Average Sales
average_sales = df["Total_Sales"].mean()
average_sales
print(f"Average Sales: ₹{average_sales:.2f}")

Average Sales: ₹4472.63


In [47]:
# Highest Sales
highest_sales = df["Total_Sales"].max()
highest_sales
print(f"Highest Sales: ₹{highest_sales:.2f}")

Highest Sales: ₹13995.00


In [49]:
# Lowest Sales
lowest_sales = df["Total_Sales"].min()
lowest_sales
print(f"Lowest Sales: ₹{lowest_sales:.2f}")

Lowest Sales: ₹449.10


In [100]:
# Total Quantity Sold
total_quantity = df["Quantity"].sum()
total_quantity
print(f"Total Quantity Sold: {total_quantity}")

Total Quantity Sold: 294


In [55]:
# Number of Orders
number_of_orders = df["Order_ID"].nunique()
number_of_orders
print("Number of Orders:", number_of_orders)

Number of Orders: 100


In [57]:
# Highest-selling order
df.loc[df["Total_Sales"].idxmax()]

Order_ID                     1049
Order_Date             2026-03-02
Customer             Ishita Gupta
City                       Mumbai
Region                       West
Category                   Sports
Product             Running Shoes
Quantity                        5
Unit_Price                   2799
Discount_Percent                0
Total_Sales               13995.0
Rating                        4.0
Payment_Method        Net Banking
Name: 48, dtype: object

In [58]:
# Lowest-selling order
df.loc[df["Total_Sales"].idxmin()]

Order_ID                        1006
Order_Date                2026-01-21
Customer                   Manya Rao
City                           Kochi
Region                         South
Category                       Books
Product                The Alchemist
Quantity                           1
Unit_Price                       499
Discount_Percent                  10
Total_Sales                    449.1
Rating                           4.2
Payment_Method      Cash on Delivery
Name: 5, dtype: object

In [97]:
print("Total Sales:", total_sales)
print("Average Sales:", average_sales)
print("Highest Sales:", highest_sales)
print("Lowest Sales:", lowest_sales)
print("Total Quantity Sold:", total_quantity)
print("Number of Orders:", number_of_orders)

Total Sales: 447262.95000000007
Average Sales: 4472.629500000001
Highest Sales: 13995.0
Lowest Sales: 449.1
Total Quantity Sold: 294
Number of Orders: 100


## 7. Category-wise Sales Analysis

#### Total sales by category
The total sales for each category are calculated using
`groupby()` and the `sum()` aggregation function.

In [103]:
category_sales = df.groupby("Category")["Total_Sales"].sum()
category_sales

Category
Books              36792.40
Clothing           69028.20
Electronics        92086.00
Home & Kitchen    113329.90
Sports            136026.45
Name: Total_Sales, dtype: float64

In [60]:
# Average sales by category
category_avg_sales = df.groupby("Category")["Total_Sales"].mean()
category_avg_sales

Category
Books             1752.019048
Clothing          4930.585714
Electronics       4846.631579
Home & Kitchen    5396.661905
Sports            5441.058000
Name: Total_Sales, dtype: float64

In [61]:
# Total quantity sold by category
category_quantity = df.groupby("Category")["Quantity"].sum()
category_quantity

Category
Books             60
Clothing          39
Electronics       57
Home & Kitchen    59
Sports            79
Name: Quantity, dtype: int64

In [62]:
# Number of orders by category
category_orders = df.groupby("Category")["Order_ID"].count()
category_orders

Category
Books             21
Clothing          14
Electronics       19
Home & Kitchen    21
Sports            25
Name: Order_ID, dtype: int64

In [63]:
category_analysis = df.groupby("Category").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
)

category_analysis

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Category,,,,
Books,36792.40,1752.019048,60,21
Clothing,69028.20,4930.585714,39,14
Electronics,92086.00,4846.631579,57,19
Home & Kitchen,113329.90,5396.661905,59,21
Sports,136026.45,5441.058000,79,25


In [65]:
# Highest-performing category
category_analysis.sort_values(by="Total_Sales", ascending=False).head(1)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Category,,,,
Sports,136026.45,5441.058,79,25


## 8. City-wise Sales Analysis

#### Total Sales by City

This analysis compares the total sales generated by
customers from different cities.

In [104]:
city_sales = df.groupby("City")["Total_Sales"].sum()
city_sales

City
Ahmedabad      14788.70
Bengaluru      13524.30
Bhubaneswar     6598.00
Chandigarh     34539.65
Chennai        47493.20
Delhi          30882.80
Hyderabad      59076.45
Jaipur         25534.00
Jammu          16059.75
Kochi          37257.50
Kolkata        20169.00
Lucknow        46194.65
Mumbai         41457.80
Pune           41908.55
Srinagar       11778.60
Name: Total_Sales, dtype: float64

In [67]:
# Average sales by city
city_avg_sales = df.groupby("City")["Total_Sales"].mean()
city_avg_sales

City
Ahmedabad      2464.783333
Bengaluru      3381.075000
Bhubaneswar    6598.000000
Chandigarh     4317.456250
Chennai        4749.320000
Delhi          5147.133333
Hyderabad      8439.492857
Jaipur         3647.714286
Jammu          2676.625000
Kochi          4139.722222
Kolkata        2881.285714
Lucknow        7699.108333
Mumbai         5182.225000
Pune           4190.855000
Srinagar       2355.720000
Name: Total_Sales, dtype: float64

In [68]:
# Total quantity sold by city
city_quantity = df.groupby("City")["Quantity"].sum()
city_quantity

City
Ahmedabad      12
Bengaluru      14
Bhubaneswar     2
Chandigarh     22
Chennai        29
Delhi          18
Hyderabad      25
Jaipur         17
Jammu          11
Kochi          29
Kolkata        24
Lucknow        22
Mumbai         23
Pune           34
Srinagar       12
Name: Quantity, dtype: int64

In [69]:
# Number of orders by city
city_orders = df.groupby("City")["Order_ID"].count()
city_orders

City
Ahmedabad       6
Bengaluru       4
Bhubaneswar     1
Chandigarh      8
Chennai        10
Delhi           6
Hyderabad       7
Jaipur          7
Jammu           6
Kochi           9
Kolkata         7
Lucknow         6
Mumbai          8
Pune           10
Srinagar        5
Name: Order_ID, dtype: int64

In [70]:
city_analysis = df.groupby("City").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
)

city_analysis

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
City,,,,
Ahmedabad,14788.70,2464.783333,12,6
Bengaluru,13524.30,3381.075000,14,4
Bhubaneswar,6598.00,6598.000000,2,1
Chandigarh,34539.65,4317.456250,22,8
Chennai,47493.20,4749.320000,29,10
Delhi,30882.80,5147.133333,18,6
Hyderabad,59076.45,8439.492857,25,7
Jaipur,25534.00,3647.714286,17,7
Jammu,16059.75,2676.625000,11,6


In [74]:
# Find the highest-performing city 
city_analysis.sort_values(by="Total_Sales", ascending=False).head(1)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
City,,,,
Hyderabad,59076.45,8439.492857,25,7


In [75]:
 # Lowest-performing city
city_analysis.sort_values(by="Total_Sales", ascending=True).head(1)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
City,,,,
Bhubaneswar,6598.0,6598.0,2,1


## 9. Product-wise Sales Analysis

In [76]:
# Total sales by product
product_sales = df.groupby("Product")["Total_Sales"].sum()
product_sales

Product
Atomic Habits              11590.65
Coffee Maker               51610.25
Cricket Bat                41483.40
Data Science Handbook       4495.00
Dumbbell Set               30184.90
Electric Kettle            26382.40
Football                   10307.10
Hoodie                      7675.20
Jacket                     47106.15
Jeans                       6456.60
Machine Learning Basics     5694.30
Mechanical Keyboard        26489.40
Mixer Grinder              25341.55
Power Bank                  6474.60
Python Programming          7527.45
Running Shoes              41285.25
Smart Watch                15670.25
T-Shirt                     7790.25
The Alchemist               7485.00
Water Bottle                9995.70
Wireless Headphones        33877.40
Wireless Mouse              9574.35
Yoga Mat                   12765.80
Name: Total_Sales, dtype: float64

In [77]:
# Total quantity sold by product
product_quantity = df.groupby("Product")["Quantity"].sum()
product_quantity

Product
Atomic Habits              21
Coffee Maker               16
Cricket Bat                18
Data Science Handbook       5
Dumbbell Set               16
Electric Kettle            19
Football                   14
Hoodie                      5
Jacket                     20
Jeans                       4
Machine Learning Basics     6
Mechanical Keyboard        11
Mixer Grinder               9
Power Bank                  6
Python Programming         11
Running Shoes              15
Smart Watch                 5
T-Shirt                    10
The Alchemist              17
Water Bottle               15
Wireless Headphones        23
Wireless Mouse             12
Yoga Mat                   16
Name: Quantity, dtype: int64

In [78]:
# Average sales by product
product_avg_sales = df.groupby("Product")["Total_Sales"].mean()
product_avg_sales

Product
Atomic Habits              1931.775000
Coffee Maker               7372.892857
Cricket Bat                8296.680000
Data Science Handbook      4495.000000
Dumbbell Set               6036.980000
Electric Kettle            5276.480000
Football                   2576.775000
Hoodie                     3837.600000
Jacket                     9421.230000
Jeans                      3228.300000
Machine Learning Basics    1423.575000
Mechanical Keyboard        8829.800000
Mixer Grinder              6335.387500
Power Bank                 2158.200000
Python Programming         1881.862500
Running Shoes              8257.050000
Smart Watch                5223.416667
T-Shirt                    1558.050000
The Alchemist              1247.500000
Water Bottle               1999.140000
Wireless Headphones        4839.628571
Wireless Mouse             3191.450000
Yoga Mat                   2127.633333
Name: Total_Sales, dtype: float64

In [79]:
# Number of orders for each product
product_orders = df.groupby("Product")["Order_ID"].count()
product_orders

Product
Atomic Habits              6
Coffee Maker               7
Cricket Bat                5
Data Science Handbook      1
Dumbbell Set               5
Electric Kettle            5
Football                   4
Hoodie                     2
Jacket                     5
Jeans                      2
Machine Learning Basics    4
Mechanical Keyboard        3
Mixer Grinder              4
Power Bank                 3
Python Programming         4
Running Shoes              5
Smart Watch                3
T-Shirt                    5
The Alchemist              6
Water Bottle               5
Wireless Headphones        7
Wireless Mouse             3
Yoga Mat                   6
Name: Order_ID, dtype: int64

In [80]:
product_analysis = df.groupby("Product").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
)

product_analysis

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Product,,,,
Atomic Habits,11590.65,1931.775000,21,6
Coffee Maker,51610.25,7372.892857,16,7
Cricket Bat,41483.40,8296.680000,18,5
Data Science Handbook,4495.00,4495.000000,5,1
Dumbbell Set,30184.90,6036.980000,16,5
Electric Kettle,26382.40,5276.480000,19,5
Football,10307.10,2576.775000,14,4
Hoodie,7675.20,3837.600000,5,2
Jacket,47106.15,9421.230000,20,5


In [81]:
# Top 3 products by quantity sold
product_analysis.sort_values(
    by="Total_Quantity",
    ascending=False
).head(3)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Product,,,,
Wireless Headphones,33877.40,4839.628571,23,7
Atomic Habits,11590.65,1931.775000,21,6
Jacket,47106.15,9421.230000,20,5


In [82]:
# Lowest 3 products by sales
product_analysis.sort_values(
    by="Total_Sales",
    ascending=True
).head(3)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Product,,,,
Data Science Handbook,4495.0,4495.000,5,1
Machine Learning Basics,5694.3,1423.575,6,4
Jeans,6456.6,3228.300,4,2


## 10. Payment Method Analysis

In [83]:
# Total sales by payment method
payment_sales = df.groupby("Payment_Method")["Total_Sales"].sum()
payment_sales

Payment_Method
Cash on Delivery     50123.30
Credit Card         106540.25
Debit Card           98064.90
Net Banking          96371.25
UPI                  96163.25
Name: Total_Sales, dtype: float64

In [84]:
# Number of orders by payment method
payment_orders = df.groupby("Payment_Method")["Order_ID"].count()
payment_orders

Payment_Method
Cash on Delivery    15
Credit Card         22
Debit Card          20
Net Banking         17
UPI                 26
Name: Order_ID, dtype: int64

In [85]:
# Total quantity by payment method
payment_quantity = df.groupby("Payment_Method")["Quantity"].sum()
payment_quantity

Payment_Method
Cash on Delivery    32
Credit Card         65
Debit Card          64
Net Banking         63
UPI                 70
Name: Quantity, dtype: int64

In [86]:
payment_analysis = df.groupby("Payment_Method").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
)

payment_analysis

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Payment_Method,,,,
Cash on Delivery,50123.30,3341.553333,32,15
Credit Card,106540.25,4842.738636,65,22
Debit Card,98064.90,4903.245000,64,20
Net Banking,96371.25,5668.897059,63,17
UPI,96163.25,3698.586538,70,26


In [88]:
# Most used payment method
payment_analysis.sort_values(
    by="Number_of_Orders",
    ascending=False
).head(1)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Payment_Method,,,,
UPI,96163.25,3698.586538,70,26


In [89]:
# Payment method with highest sales
payment_analysis.sort_values(
    by="Total_Sales",
    ascending=False
).head(1)

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Payment_Method,,,,
Credit Card,106540.25,4842.738636,65,22


## 11. Top Performing Orders

In [90]:
# Top 5 orders by sales
top_orders = df.sort_values(
    by="Total_Sales",
    ascending=False
).head(5)

top_orders[["Order_ID", "Customer", "Product", "Category", "Total_Sales"]]

,Order_ID,Customer,Product,Category,Total_Sales
48,1049,Ishita Gupta,Running Shoes,Sports,13995.00
52,1053,Ananya Singh,Running Shoes,Sports,13295.25
84,1085,Sana Malik,Coffee Maker,Home & Kitchen,12596.40
89,1090,Aditya Verma,Cricket Bat,Sports,12495.00
12,1013,Maryam Khan,Jacket,Clothing,10620.75


In [91]:
# Top 5 products by sales
top_products = product_analysis.sort_values(
    by="Total_Sales",
    ascending=False
).head(5)

top_products

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Product,,,,
Coffee Maker,51610.25,7372.892857,16,7
Jacket,47106.15,9421.230000,20,5
Cricket Bat,41483.40,8296.680000,18,5
Running Shoes,41285.25,8257.050000,15,5
Wireless Headphones,33877.40,4839.628571,23,7


In [92]:
# Top 3 categories by sales
top_categories = category_analysis.sort_values(
    by="Total_Sales",
    ascending=False
).head(3)

top_categories

,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Category,,,,
Sports,136026.45,5441.058000,79,25
Home & Kitchen,113329.90,5396.661905,59,21
Electronics,92086.00,4846.631579,57,19


In [93]:
# Orders with sales above ₹5,000
high_value_orders = df[df["Total_Sales"] > 5000]

high_value_orders[[
    "Order_ID",
    "Customer",
    "Product",
    "Category",
    "Total_Sales"
]]

,Order_ID,Customer,Product,Category,Total_Sales
3,1004,Kabir Ali,Mechanical Keyboard,Electronics,9496.20
7,1008,Aman Kumar,Hoodie,Clothing,6076.20
9,1010,Aman Kumar,Jacket,Clothing,8996.40
10,1011,Rahul Das,Coffee Maker,Home & Kitchen,6998.00
12,1013,Maryam Khan,Jacket,Clothing,10620.75
13,1014,Dev Patel,Running Shoes,Sports,8397.00
16,1017,Sana Malik,Cricket Bat,Sports,7497.00
18,1019,Reyansh Jain,Jacket,Clothing,7497.00
27,1028,Sana Malik,Wireless Headphones,Electronics,5696.20
28,1029,Sara Ahmed,Wireless Headphones,Electronics,5696.20


In [94]:
# Highly rated orders with high sales
df[(df["Rating"] >= 4.5) & (df["Total_Sales"] > 5000)][
    ["Order_ID", "Product", "Rating", "Total_Sales"]
]

,Order_ID,Product,Rating,Total_Sales
9,1010,Jacket,4.7,8996.40
12,1013,Jacket,4.8,10620.75
13,1014,Running Shoes,4.6,8397.00
18,1019,Jacket,4.7,7497.00
27,1028,Wireless Headphones,4.6,5696.20
32,1033,Mixer Grinder,4.7,8097.30
37,1038,Dumbbell Set,5.0,9495.25
40,1041,Coffee Maker,4.6,6648.10
47,1048,Coffee Maker,5.0,10497.00
56,1057,Coffee Maker,4.5,8922.45


## 📊 Overall Sales Summary

In [105]:
summary = pd.DataFrame({
    "Metric": [
        "Total Sales",
        "Average Sales",
        "Highest Sales",
        "Lowest Sales",
        "Total Quantity Sold",
        "Number of Orders"
    ],
    "Value": [
        f"₹{total_sales:.2f}",
        f"₹{average_sales:.2f}",
        f"₹{highest_sales:.2f}",
        f"₹{lowest_sales:.2f}",
        total_quantity,
        number_of_orders
    ]
})

summary

,Metric,Value
0,Total Sales,₹447262.95
1,Average Sales,₹4472.63
2,Highest Sales,₹13995.00
3,Lowest Sales,₹449.10
4,Total Quantity Sold,294
5,Number of Orders,100


## 12. 📝 Final Observations and Insights

### Key Findings

- The dataset contains 100 orders with total sales of ₹447,262.95. The average sales value per order is approximately ₹4,472.63, and a total of 294 units were sold.

- The highest individual order has a sales value of ₹13,995, while the lowest individual order has a sales value of ₹449.10.

- Among all categories, Sports is the top-performing category with total sales of ₹136,026.45. Home & Kitchen is the second-highest category with sales of ₹113,329.90.

- Sports also has the highest total quantity sold, with 79 units across 25 orders. This indicates that Sports products have a strong contribution to both sales and order volume.

- Hyderabad recorded the highest total sales among the cities, with ₹59,076.45. Lucknow also showed a relatively high average sales value of ₹7,699.11 per order.

- Coffee Maker is the highest-selling product by total sales, generating ₹51,610.25 from 7 orders. Jacket is another strong-performing product with total sales of ₹47,106.15.

- Some products have a high average sales value even with fewer orders. For example, Jacket has an average sales value of ₹9,421.23, while Mechanical Keyboard has an average of ₹8,829.80.

- Credit Card is the payment method with the highest total sales at ₹106,540.25. UPI has the highest number of orders with 26 orders, showing that it is the most frequently used payment method in the dataset.

- Net Banking has the highest average sales value among the payment methods, at approximately ₹5,668.90 per order.

- The filtering and sorting operations helped identify high-value orders, top-performing products, categories, cities, and payment methods.

### Conclusion

Overall, the analysis shows that sales performance varies across categories, cities, products, and payment methods. Sports and Home & Kitchen are the strongest categories, while Coffee Maker and Jacket are among the top-performing products. Hyderabad recorded the highest city-wise sales, and Credit Card generated the highest total sales among payment methods. Pandas functions such as filtering, sorting, `groupby()`, and aggregation were useful for finding these patterns and understanding the e-commerce sales data.